packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelLoweringExtension.kt

```diff
@@ -57,6 +57,10 @@ private class RealmModelLowering(private val pluginContext: IrPluginContext) : C
             if (irClass.isData) {
                 error("Data class '${irClass.kotlinFqName}' is not currently supported.")
             }
+            // We don't support enum class
+            if (irClass.isEnumClass) {
```

Comment:
I know this is outside the scope of this PR, but since it was this easy to add. Could we also add a check for `object Foo : RealmObject` here? (Singleton objects)